components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -234,6 +236,35 @@ public Application updateApplication(ApplicationUpdateRequest updateRequest, Str
                 String backChannelLogoutUri = validateBackchannelLogoutURI(updateRequest.getBackchannelLogoutUri());
                 appDTO.setBackChannelLogoutUrl(backChannelLogoutUri);
             }
+            if (CollectionUtils.isNotEmpty(updateRequest.getAud())) {
+                appDTO.setAudiences(updateRequest.getAud().toArray(new String[0]));
+            }
+
+            if (updateRequest.getIdTokenEncryptionAlgorithm() != null ||
+                    updateRequest.getIdTokenEncryptionMethod() != null) {
+                appDTO.setIdTokenEncryptionEnabled(true);
+                if (updateRequest.getIdTokenEncryptionAlgorithm() != null) {
+                    appDTO.setIdTokenEncryptionAlgorithm(updateRequest.getIdTokenEncryptionAlgorithm());
+                }
+                if (updateRequest.getIdTokenEncryptionMethod() != null) {
+                    appDTO.setIdTokenEncryptionMethod(updateRequest.getIdTokenEncryptionMethod());
+                }
+                if (appDTO.getIdTokenEncryptionAlgorithm() == null &&
+                        appDTO.getIdTokenEncryptionMethod() != null) {
+                    throw DCRMUtils.generateClientException(DCRMConstants.
+                            ErrorMessages.BAD_REQUEST_INSUFFICIENT_DATA, null);
+                }
+                if (appDTO.getIdTokenEncryptionAlgorithm() != null &&
+                        appDTO.getIdTokenEncryptionMethod() == null) {
+                    appDTO.setIdTokenEncryptionMethod(OAuthServerConfiguration.getInstance().
+                            getDefaultIdTokenEncryptionMethod());
+                }
```

Comment:
What is the reason for setting the default value in second if condition and throwing the error in first if condition?

Is it okay to send an IdTokenEncryptionMethod without a IdTokenEncryptionAlgorithm and not the other way?

Reply:
According to specs, If user sends an `idTokenEncryptionAlgorithm` without `idTokenEncryptionMethod` we should set `idTokenEncryptionMethod` to the default value. If user send `idTokenEncryptionMethod` without `idTokenEncryptionAlgorithm` we should throw an error.
[https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata](https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata)

Reply:
IMO it is better if we can include this information as a comment in the code.